assets/scripts/script.js

```diff
@@ -0,0 +1,125 @@
+// defines selected elements in varibales for reuse throughout this script
+let weatherIcon = document.querySelector(".weather-icon");
+let rainfallFigure = document.querySelector(".rainfall-figure");
+let weatherIconContainer = document.querySelector(".weather-icon-container");
+let rainfallContainer = document.querySelector(".rainfall-icon-container");
+
+let rainfallFigureContent = rainfallFigure.textContent;
+let weatherIconSource = weatherIcon.src;
+
+const weatherApiKey = "7e5922ef7f6bc85e485e53b28667f43a";
+// const weatherApiUrlNormal = `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lon}&exclude=${excludeParams}&appid=${weatherApiKey}&units=metric`
+// const weatherApiUrlRain = `https://api.openweathermap.org/data/2.5/onecall/timemachine?lat=${lat}&lon=${lon}&exclude=${excludeParams}&appid=${weatherApiKey}&units=metric&dt=${yesterdayUnix}`
+
+function updateWeatherWidgets() {
+    var options = {
+        enableHighAccuracy: true,
+        timeout: 5000,
+        maximumAge: 0,

```

Comment:
What does the maximumAge  in options do?

Reply:
this is a parameter of the geolocation api call. 

more info can be found here: https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition#parameters:~:text=maximumAge%3A%20integer%20(milliseconds)%20%7C%20infinity%20%2D%20maximum%20cached%20position%20age.